Demos/Priority/Allocator/PriorityAllocator.py

```diff
@@ -174,9 +188,12 @@ def allocate(self, agents: List["Agent"], environment: "Environment", tick: int)
         allocations: Dict["Agent", "Allocation"] = {}
         displacements: Dict["Agent", Set["Agent"]] = {}
         agents_to_allocate = set(agents)
-        while len(list(agents_to_allocate)) > 0:
+        while len(agents_to_allocate) > 0:
             start_time = time_ns()
-            agent = max(agents_to_allocate, key=lambda _agent: self.priority(_agent, tick, environment))
+            max_prio = max([self.priority(_agent, tick, environment) for _agent in agents_to_allocate])
+            agent = max(
+                [_agent for _agent in agents_to_allocate if self.priority(_agent, tick, environment) == max_prio],
+                key=lambda _agent: hash(_agent))
```

Comment:
why do this twice?

Reply:
um die resultate reproduzierbar zu halten werden agents erst nach ihrer prio sortiert und dann nach ihrem hash. so ist die sortierung innerhalb eines owners immer gleich.